src/main/java/seedu/multitasky/model/ModelManager.java

```diff
@@ -62,72 +61,69 @@ public ModelManager(ReadOnlyEntryBook entryBook, UserPrefs userPrefs) {
+        filteredFloatingTaskList = new FilteredList<>(this.entryBook.getFloatingTaskList());
     }
-    // @@author
 
     public ModelManager() {
         this(new EntryBook(), new UserPrefs());
     }
 
+    // =========== EntryBook-level Operations ===========
+
     @Override
     public void resetData(ReadOnlyEntryBook newData) {
-        _entryBook.resetData(newData);
+        entryBook.resetData(newData);
         indicateEntryBookChanged();
     }
 
     @Override
     public ReadOnlyEntryBook getEntryBook() {
-        return _entryBook;
+        return entryBook;
     }
 
-    // =========== List Level Operations ===========
-
-    // @@author A0126623L
-
-    @Override
-    public void clearStateSpecificEntries(Entry.State state) {
-        _entryBook.clearStateSpecificEntries(state);
-        indicateEntryBookChanged();
-    }
-    // @@author
-
     // =========== Entry Level Operations ===========
 
     @Override
-    public synchronized void deleteEntry(ReadOnlyEntry target)
-            throws EntryNotFoundException {
-        _entryBook.removeEntry(target);
-
-        indicateEntryBookChanged();
-    }
-
-    // @@author A0126623L
-    @Override
     public synchronized void addEntry(ReadOnlyEntry entry)
             throws DuplicateEntryException, OverlappingEventException,
             OverlappingAndOverdueEventException, EntryOverdueException {
         try {
-            _entryBook.addEntry(entry);
+            entryBook.addEntry(entry);
         } catch (OverlappingEventException | OverlappingAndOverdueEventException
                  | EntryOverdueException e) {
             indicateEntryBookChanged();
             throw e;
         }
         indicateEntryBookChanged();
     }
-    // @@author
 
-    // @@author A0126623L
     @Override
-    public void changeEntryState(ReadOnlyEntry entryToChange, Entry.State newState)
+    public synchronized void deleteEntry(ReadOnlyEntry target)
+            throws EntryNotFoundException {
+        entryBook.removeEntry(target);
+
+        indicateEntryBookChanged();
+    }
+
+    @Override
+    public void updateEntry(ReadOnlyEntry target, ReadOnlyEntry editedEntry)
             throws DuplicateEntryException, EntryNotFoundException, OverlappingEventException,
             OverlappingAndOverdueEventException, EntryOverdueException {
+        requireAllNonNull(target, editedEntry);
         try {
-            _entryBook.changeEntryState(entryToChange, newState);
+            if (target.getClass().equals(editedEntry.getClass())) { // updating to same type of entry
+                entryBook.updateEntry(target, editedEntry);
+            } else { // updating to a different type of entry
+                /**
+                 * Adding is done before removal because adding may fail,
+                 * in which case removal should not be carried out.
+                 */
+                entryBook.addEntry(editedEntry);
+                entryBook.removeEntry(target);
+            }
```

Comment:
i'm not sure whether that algorithm will work or not, can u try editing from float to deadline or to event and see what happens then?